gensim/api/__init__.py

```diff
@@ -0,0 +1,174 @@
+                is_installed = True
+                sys.exit(0)
+    if os.path.exists(data_folder_dir) and not is_installed:
+        for line in f:
+            if downloaded_message in line:
+                is_downloaded = True
+                break
+    if not is_downloaded:
+        if not os.path.exists(data_folder_dir):
+            logger.info("Creating %s", data_folder_dir)
+            os.makedirs(data_folder_dir)
+            if os.path.exists(data_folder_dir):
+                logger.info("Creation of %s successful.", data_folder_dir)
+            else:
+                logger.error(
+                    "Not able to create %s. Make sure you have the correct read/"
+                    "write permissions for %s or you can try creating it manually",
+                    data_folder_dir, base_dir)
+                sys.exit(0)
+        logger.info("Downloading %s", dataset)
+        urllib.urlretrieve(url, compressed_folder_dir)
+        data_url = data_links(dataset)
+        if data_url is not None:
+            index = data_url.rfind("/")
+            data_dir = os.path.join(data_folder_dir, data_url[index + 1:])
+            urllib.urlretrieve(data_url, data_dir)
+        logger.info("%s downloaded", dataset)
+    if not is_installed:
+            tar = tarfile.open(compressed_folder_dir)
+            logger.info("Extracting files from %s", data_folder_dir)
+            tar.extractall(data_folder_dir)
+            tar.close()
+            logger.info("%s installed", dataset)
+
+
+def catalogue(print_list=False):
+    url = "https://raw.githubusercontent.com/chaitaliSaini/Corpus_and_models/master/list_with_filename.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    if print_list:
+        corpuses = data['gensim']['corpus']
+        models = data['gensim']['model']
+        print("Corpuses available : ")
+        for corpus in corpuses:
+            print(corpus)
+        print("")
+        print("Models available : ")
+        for model in models:
+            print(model)
+    return data
+
+
+def info(dataset):
+    data = catalogue()
+    corpuses = data['gensim']['corpus']
+    models = data['gensim']['model']
+    if dataset in corpuses:
+        print(data['gensim']['corpus'][dataset]["desc"])
+    elif dataset in models:
+        print(data['gensim']['model'][dataset]["desc"])
+    else:
+        catalogue(print_list=True)
+        raise Exception(
+            "Incorrect model/corpus name. Choose the model/corpus from the list "
+            "above.")
+
+
+def get_filename(dataset):
+    data = catalogue()
+    corpuses = data['gensim']['corpus']
+    models = data['gensim']['model']
+    if dataset in corpuses:
+        return data['gensim']['corpus'][dataset]["filename"]
+    elif dataset in models:
+        return data['gensim']['model'][dataset]["filename"]
+
+
+def load(dataset, return_path=False):
```

Comment:
Please add doc-strings to all functions in [google-style format](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) (here and anywhere).

Reply:
Also, please merge  `load` and `download` to new `load` function (If I have no needed model on my PC, I want to download it, not  see an exception).